nativeApp/src/lib/ui/forms/TimeForm/index.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { FC, useState } from 'react';
+import styled from 'styled-components/native';
+import { Form as F, Text } from 'native-base';
+
+import Dropdown from '../Dropdown';
+import { Forms } from '../enums';
+import DateTimePicker from '../DateTimePicker';
+import { ColoursEnum } from '../../colours';
+import HoursAndMinutesText from '../../HoursAndMinutesText';
+import SubmitButton from '../SubmitButton';
+
+import { getTimeDiff } from '../../../utils/time';
+
+
+/*
+ * Types
+ */
+export type IdAndName = { id: number; name: string }
+export type TimeValues = {
+  project: string;
+  activity: string;
+  volunteer: string;
+  date: Date;
+  startTime: Date;
+  endTime: Date;
+
+}
+type Props = {
+  onSubmit: (v: TimeValues) => void;
+  initialValues?: TimeValues;
+  activities: IdAndName[];
+  projects: IdAndName[];
+  volunteers?: IdAndName[];
+  forUser: 'admin' | 'volunteer';
+}
+
+/*
+ * Styles
+ */
+const Form = styled(F)`
+  width: ${Forms.formWidth}
+`;
+
+const Label = styled(Text)`
+  color: ${ColoursEnum.darkGrey};
+  marginTop: 15;
+`;
+
+const TimeContainer = styled.View`
+  marginTop: 15;
+`;
+
+/*
+ * Component
+ */
+const TimeForm: FC<Props> = (props) => {
+  const [project, setProject] = useState('');
+  const [activity, setActivity] = useState('');
+  const [volunteer, setVolunteer] = useState('');
+  const [date, setDate] = useState<Date>();
+  const [startTime, setStartTime] = useState<Date>();
+  const [endTime, setEndTime] = useState<Date>();
+
+  const diff = getTimeDiff(startTime, endTime);
+
+  const {
+    forUser, activities, projects, volunteers,
+  } = props;
+  return (
+    <Form>
+      {forUser === 'admin' && <Dropdown label="Volunteer" options={volunteers} selectedValue={volunteer} onValueChange={setVolunteer} />}
```

Comment:
I was unaware of the multi select. I may defer this to later in the build if that's okay as it will require a fair bit of work to match the dropdown styling